packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -882,10 +883,23 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     tsParseTupleElementType(): N.TsType | N.TsNamedTupleMember {
       // parses `...TsType[]`
 
-      const { start: startPos, startLoc } = this.state;
+      const { start: startPos, startLoc, value } = this.state;
+
+      const { keyword } = this.state.type;
 
       const rest = this.eat(tt.ellipsis);
-      let type = this.tsParseType();
+      let type;
+      const ALLOWED_KEYWORDS = ["function"];
```

Comment:
IMO TS should have allowed all keywords as the label in the named tuple members, I have file an issue to TS: https://github.com/microsoft/TypeScript/issues/45819

After that issue is resolved, we can remove the `ALLOWED_KEYWORDS` here.

Reply:
The TS issue has been marked as [suggestion]/[awaiting more feedback]. Maybe we can start by matching TS's behavior, and then relax the restriction when they relax it.

Reply:
If we want to match the TS behaviour, which I think is an overlook when implementing named tuple elements, the complete "allowed" keyword set can be found in https://github.com/microsoft/TypeScript/blob/0af2497fecef9e41d7d0260fd37932dd33912c66/src/compiler/parser.ts#L3726

Reply:
I 100% agree that this is an overlook, but the TS team doesn't seem to much interested in relaxing this restriction :sweat_smile: 

As an alternative, we can just allow every keyword: valid TS programs will still be accepted.